.travis.yml

```diff
@@ -147,13 +192,30 @@ script:
         else
             echo skip
         fi
-  - if [ -n "$SCA" ]; then pyflakes . || exit; else echo skip; fi
+  - if [ "$MODE" == 'SCA' ]; then pyflakes . || exit; else echo skip; fi
 
   # isort doesn't get along with yapf. Hopefully we'll figure it out and re-enable in the future.
-  # - if [ -n "$SCA" ]; then isort --order-by-type --recursive --line-width 100 --diff --verbose --check-only || exit; else echo skip; fi
+  # - if [ "$MODE" == 'SCA' ]; then isort --order-by-type --recursive --line-width 100 --diff --verbose --check-only || exit; else echo skip; fi
 
   # home-made bad code detectors
-  - if [ -n "$SCA" ]; then grep -r --extended-regex '^ *class [^\(]+(\(\))?:' . && exit 1 || true; else echo skip; fi  # old-style class detector
-  - if [ -n "$SCA" ]; then grep -r 'isinstance\(' . && exit 1 || true; else echo skip; fi  # very non-pythonic
-  - if [ -n "$SCA" ]; then grep -r --extended-regex '\b(all|any)\(\[' . && exit 1 || true; else echo skip; fi  # a list is unnecesarily allocated where a generator expression could have been used
-  - if [ -n "$SCA" ]; then grep -r '\_\_metaclass\_\_' . && exit 1 || true; else echo skip; fi  # bad metaclass declaration: use six
+  - if [ "$MODE" == 'SCA' ]; then grep -r --extended-regex '^ *class [^\(]+(\(\))?:' . && exit 1 || true; else echo skip; fi  # old-style class detector
```

Comment:
In some parts of the code above you use `egrep` and here you use `grep --extended-regex`. I think it could be more readable if you choose one way and use it consequently.